integration/src/main/java/org/apache/mahout/utils/clustering/JsonClusterWriter.java

```diff
@@ -70,20 +71,30 @@ public void write(ClusterWritable clusterWritable) throws IOException {
     Map<String, Object> res = Maps.newHashMap();
 
     // get top terms
-    List<Object> topTerms = getTopFeaturesList(clusterWritable.getValue()
+    if (dictionary != null) {
+      List<Object> topTerms = getTopFeaturesList(clusterWritable.getValue()
         .getCenter(), dictionary, numTopFeatures);
-    res.put("top_terms", topTerms);
-
+      res.put("top_terms", topTerms);
+    } else {
+      res.put("top_terms", Lists.newArrayList());
```

Comment:
Replace Lists.newArrayList() with new ArrayList<>(). Since we are now on Java 7. Also replace all similar codes in this code.  Same with Maps.newHashMap() to be replaced by new HashMap<>()